### PR TITLE
Paginated views

### PR DESCRIPTION
**Pull request by JFer11**
In this branch, pagination fir existing endpoints was added.
Two pagination methods were implemented, and one of them was slightly modified to accept another parameter in the `request header`.
Said parameter was how we want the results, either ascending or descending way.

### CODE REVIEW

File: articles/app_articles/paginations.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from rest_framework.pagination import LimitOffsetPagination, PageNumberPagination
+
+
+class MyPaginationArticles(LimitOffsetPagination):
```
Comment (joaquinco) [inline articles/app_articles/paginations.py]: Maybe `ArticlesPagination` is more suitable

File: articles/app_articles/paginations.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        order = self.get_asc_or_desc(request)
+        if order == 'DESC':
+            queryset = queryset.order_by('-created')
+        elif order == 'ASC':
+            queryset = queryset.order_by('created')
+        elif order is None:
+            # No order requested
+            pass
+
+        return list(queryset[self.offset:self.offset + self.limit])
```
Comment (joaquinco) [inline articles/app_articles/paginations.py]: I don't think this offers any adventage over the super class implementation.

I think i'd be better to move the ordering logic to a filter class or to the view

Comment (JFer11) [reply articles/app_articles/paginations.py]: I was following the suggested implementation in Trello:
"Allow to sort articles by: - creation date desc / asc"
So I modified the super class pagination behavior a bit. 
What do you recommend?

Comment (JFer11) [reply articles/app_articles/paginations.py]: I changed it. Now the filter is done by get_queryset instead of the paging class.

File: articles/app_articles/views/article_views.py [hunk 3]
```diff
@@ -11,6 +15,36 @@ class ArticleViewSet(viewsets.ModelViewSet):
     queryset = Article.objects.all()
     serializer_class = ArticleSerializer
     permission_classes = [IsAuthenticated]
+    pagination_class = ArticlesPagination
+
+    def get_asc_or_desc(self, request, queryset):
+        if request is None:
+            raise NullRequest
```
Comment (joaquinco) [inline articles/app_articles/views/article_views.py:22]: You can assume this won't happen. Also, since this is an instance method, you can use `self.request`

File: articles/app_articles/views/article_views.py [hunk 4]
```diff
@@ -11,6 +15,36 @@ class ArticleViewSet(viewsets.ModelViewSet):
...
+                "'%s' should either include a `queryset` attribute, "
+                "or override the `get_queryset()` method."
+                % self.__class__.__name__
+        )
+
+        queryset = self.get_asc_or_desc(self.request, self.queryset)
+
+        if isinstance(queryset, QuerySet):
+            # Ensure queryset is re-evaluated on each request.
+            queryset = queryset.all()
```
Comment (joaquinco) [inline articles/app_articles/views/article_views.py:46]: Why is this needed?

**Review by joaquinco — APPROVED** [review]

File: articles/app_articles/views/article_views.py (hunk 3 above)
Comment (JFer11) [reply articles/app_articles/views/article_views.py:22]: I put it there because maybe for some reason unknown to me, get_queryset is called from another function that does not involve an API request, so in that case it will be None.

File: articles/app_articles/views/article_views.py (hunk 4 above)
Comment (JFer11) [reply articles/app_articles/views/article_views.py:46]: It was provided in the original Django `get_queryset `implementation. I have an idea, why it is there, but I am not sure. However, i decided to not challenge Django implementation and keep it there.

Comment (joaquinco) [reply articles/app_articles/views/article_views.py:46]: Ok. For next time, a better approach would be, instead of reimplementing the same as django, calling super()
